tests/conftest.py

```diff
@@ -577,6 +577,40 @@ def rand_one_dut_lossless_prio(request):
     return lossless_prio_list[0]
 
 
+@pytest.fixture(scope="module")
+def rand_asic_namespace(duthosts, rand_one_dut_hostname):
+    """
+    Return the randomly selected asic namespace in case of multi-asic duthost.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    asic_namespace = None
+    asic_index = None
+    if duthost.is_multi_asic:
+        namespace_list = duthost.get_asic_namespace_list()
+        asic_namespace = random.choice(namespace_list)
+        asic_index = duthost.get_asic_id_from_namespace(asic_namespace)
+
+    return asic_namespace, asic_index
+
+
+@pytest.fixture(scope="module")
+def rand_front_end_asic_namespace(duthosts, rand_one_dut_front_end_hostname):
+    """
+    Return the randomly selected asic namespace in case of multi-asic frontend duthost.
+    """
+    duthost = duthosts[rand_one_dut_front_end_hostname]
+
+    asic_namespace = None
+    asic_index = None
+    if duthost.is_multi_asic:
+        namespace_list = duthost.get_asic_namespace_list()
+        asic_namespace = random.choice(namespace_list)
+        asic_index = duthost.get_asic_id_from_namespace(asic_namespace)
+
+    return asic_namespace, asic_index
```

Comment:
These two functions are identical.

Reply:
Actually, they could potentially use different duthost types. `rand_asic_namespace` can use any type of duthost, while `rand_front_end_asic_namespace` requires only front-end duthosts. This separation was necessary because some features run only on front-end duthosts in a multi-ASIC T2 platform and are not applicable to supervisor cards. Test suites are adapted accordingly based on their requirements to select the appropriate fixture, depending on when the feature is valid.

Reply:
Actually, although these two functions execute differently, the code is indeed duplicated. I will work on a solution to keep a single common function that adapts based on the availability of the two duthost fixtures.